core/controllers/base_test.py

```diff
@@ -494,3 +494,47 @@ def test_error_response_for_get_request_of_type_json_has_json_format(self):
             response = self.get_json('/fake', expect_errors=True)
             self.assertTrue(isinstance(response, dict))
             self.assertEqual(500, response['code'])
+
+
+class CheckAllHandlersHaveDecorator(test_utils.GenericTestBase):
+    """Tests that all methods in handlers have authentication decorators
+    applied on them.
+    """
+
+    def test_every_method_has_decorator(self):
+        handlers_checked = set([])
+
+        for route in main.URLS:
+            # URLS = MAPREDUCE_HANDLERS + other handers. MAPREDUCE_HANDLERS
+            # are touples. So, below check is to handle them.
+            if isinstance(route, tuple):
+                continue
+            else:
+                handler = route.handler
+
+            # Following handler are present in base.py where acl_decorators
+            # cannot be imported.
+            if (handler.__name__ == 'LogoutPage' or
+                    handler.__name__ == 'Error404Handler'):
+                continue
+
+            if handler.get != base.BaseHandler.get:
+                handlers_checked.add(handler)
+                self.assertTrue(hasattr(handler.get, '__wrapped__'))
+
+            if handler.post != base.BaseHandler.post:
+                handlers_checked.add(handler)
+                self.assertTrue(hasattr(handler.post, '__wrapped__'))
+
+            if handler.put != base.BaseHandler.put:
+                handlers_checked.add(handler)
+                self.assertTrue(hasattr(handler.put, '__wrapped__'))
+
+            if handler.delete != base.BaseHandler.delete:
+                handlers_checked.add(handler)
+                self.assertTrue(hasattr(handler.delete, '__wrapped__'))
+
```

Comment:
it's an assertion here. So the flow will stop when a case for handler fails.

Should I change it from assertion to if/else so that all cases are run and pass/fail is shown for all?